source_code/src/OLEDMINI/bitstreammini.c

```diff
@@ -45,38 +45,36 @@ void miniBistreamInit(bitstream_mini_t* bs, uint8_t height, uint16_t width, uint
     bs->width = width;
     bs->height = height;
     bs->dataCounter = 0;
-    bs->bufferInd = sizeof(bs->buffer);
-    bs->dataSize = (uint16_t)width * (((uint16_t)height+7) >> BITSTREAM_PIXELS_PER_BYTE_BITSHIFT);
+    bs->dataSize = (uint16_t)width * (((uint16_t)height+7) / BITSTREAM_PIXELS_PER_BYTE);
 }
 
 /*! \fn     bsGetNextByte(bitstream_mini_t* bs)
  *  \brief  Return the next data byte from flash
  *  \param  bs      pointer to initialized bitstream context to get the next word from
  *  \return next data word, or 0 if end of data reached
  */
-uint8_t miniBistreamGetNextByte(bitstream_mini_t* bs)
-{
-    // Are you requesting bytes when you've already read everything?
-    if (bs->dataCounter < bs->dataSize) 
-    {
-        bs->dataCounter++;
-        
-        // Check if we need to fetch new data from the SPI flash
-        if(bs->bufferInd >= sizeof(bs->buffer))
-        {
-            // Fetch new data from external flash
-            flashRawRead(bs->buffer, bs->addr, sizeof(bs->buffer));
-            bs->addr += sizeof(bs->buffer);
-            bs->bufferInd = 0;
-            //usbPrintf_P(PSTR("bistream buffer: %02x %02x %02x %02x %02x %02x"), bs->buffer[0], bs->buffer[1], bs->buffer[2], bs->buffer[3], bs->buffer[4], bs->buffer[5]);
-        }
-        
-         return bs->buffer[bs->bufferInd++];
-    }
-    else
-    {
-        return 0;
-    }
-}
+ uint8_t miniBistreamGetNextByte(bitstream_mini_t* bs)
+ {
+     // Are you requesting bytes when you've already read everything?
+     if (bs->dataCounter < bs->dataSize)
+     {
+         uint16_t buffer_index = bs->dataCounter % sizeof(bs->buffer);
```

Comment:
Did you misstype this question?

I've only said this PR improves flash usage, nothing stated about speed.
However if you are concerning speed:
* The % operations is of the power of 2. which is basically just a & 0x0F if i am not wrong. Compiler magic
* I've even tested the function without a buffer at all. This means every pixel is read for every request just in time. The speed was acceptable, but in the main menu where all pixels are updated very fast this was a tiny bit too slow.

So if your question was more about if my changes will slow down the image process I am of the opinion that it does not, or even if it would not that we'd notice it.